client/src/components/ItineraryList/compareUtils.ts

```diff
@@ -0,0 +1,86 @@
+import { TripQuery, Mode } from '../../gql/graphql.ts';
+import { formatTime } from '../../util/formatTime.ts';
+import { formatDuration } from '../../util/formatDuration.ts';
+import { formatDistance } from '../../util/formatDistance.ts';
+
+// Extract types from TripQuery for better type safety
+export type TripPatternType = NonNullable<TripQuery['trip']['tripPatterns'][0]>;
```

Comment:
These already exist in client/src/static/query/tripQueryTypes.d.ts, maybe there's a better place for them? We can also make them "global" perhaps.